cli/bpcatalog/gh.go

```diff
@@ -0,0 +1,110 @@
+package bpcatalog
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+	"sort"
+
+	"github.com/google/go-github/v47/github"
+	"golang.org/x/oauth2"
+)
+
+const ghTokenEnvVar = "GITHUB_TOKEN"
+
+type ghService struct {
+	client *github.Client
+	ctx    context.Context
+	orgs   []string
+}
+
+type ghServiceOption func(*ghService)
+
+func withOrgs(orgs []string) ghServiceOption {
+	return func(g *ghService) {
+		g.orgs = orgs
+	}
+}
+
+func withClient(c *http.Client) ghServiceOption {
+	return func(g *ghService) {
+		g.client = github.NewClient(c)
+	}
+}
+
+func withTokenClient() ghServiceOption {
+	return func(g *ghService) {
+		pat, isSet := os.LookupEnv(ghTokenEnvVar)
+		if !isSet {
+			Log.Crit(fmt.Sprintf("GitHub token env var %s is not set", ghTokenEnvVar))
+			os.Exit(1)
+		}
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: pat},
+		)
+		tc := oauth2.NewClient(g.ctx, ts)
+		g.client = github.NewClient(tc)
+	}
+}
+
+func newGHService(opts ...ghServiceOption) *ghService {
+	ctx := context.Background()
+	ghs := &ghService{
+		client: github.NewClient(nil),
+		ctx:    ctx,
+	}
+	for _, opt := range opts {
+		opt(ghs)
+	}
```

Comment:
Is this doing anything since the body for `type ghServiceOption` is empty?

Reply:
Not sure what you meant by empty - it is a type alias for func(*ghService) which can modify a struct of ghService.